infra/main.bicep

```diff
@@ -171,6 +180,9 @@ module openAi1RoleApim 'core/security/role.bicep' = {
 module openAi2RoleApim 'core/security/role.bicep' = {
   scope: resourceGroup
   name: 'openai2-role-apim'
+  dependsOn: [
```

Comment:
not needed, apim is automatically recognized as a dependency because of the param.
There's no dependency on openAi1 as the role is set at RG scope

Since the scope has change to RG, this second one role is redundant